adagrams/game.py

```diff
@@ -1,13 +1,119 @@
 from random import randint
 
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
+LETTER_SCORE = {
+    "A": 1,
+    "E": 1,
+    "I": 1,
+    "O": 1,
+    "U": 1,
+    "L": 1,
+    "N": 1,
+    "R": 1,
+    "S": 1, 
+    "T": 1,
+    "D": 2, 
+    "G": 2,
+    "B": 3, 
+    "C": 3, 
+    "M": 3, 
+    "P": 3,
+    "F": 4, 
+    "H": 4, 
+    "V": 4, 
+    "W": 4, 
+    "Y": 4,
+    "K": 5,
+    "J": 8, 
+    "X": 8,
+    "Q": 10,
+    "Z": 10
+}
+
 def draw_letters():
-    pass
+    all_letters = []
+    hand = []
+
+    for letter, count in LETTER_POOL.items():
+        for _ in range(0, count):
+            all_letters.append(letter)
+
+    for _ in range(10):
+        index = randint(0, len(all_letters) - 1)
+        hand.append(all_letters[index])
+        all_letters.pop(index)
+    return hand
 
 def uses_available_letters(word, letter_bank):
-    pass
+    bank_copy = letter_bank[:]
+
+    if len(word) > len(letter_bank):
+        return False
+        
+    for letter in word.upper():
+        if letter in bank_copy:
+            bank_copy.remove(letter)
+        else:
+            return False
+    return True
+
 
 def score_word(word):
-    pass
+    score = 0
+    upper_word = word.upper()
+
+    for letter in upper_word:
+        score += LETTER_SCORE[letter]
+
+    if len(upper_word) > 6:
+        score += 8
```

Comment:
This is another place where Magic Numbers have appeared! Try to avoid hardcoded values! We could try something like the following to fix it:

```suggestion
        BONUS_FOR_LONG_WORD = 8
        
        
        
        
        score += BONUS_FOR_LONG_WORD
```